extensions/vault/runtime/src/main/java/io/quarkus/vault/VaultTransitSecretEngine.java

```diff
@@ -0,0 +1,103 @@
+import io.quarkus.vault.transit.RewrappingRequest;
+import io.quarkus.vault.transit.SignatureInput;
+import io.quarkus.vault.transit.SigningBatchRequest;
+import io.quarkus.vault.transit.TransitContext;
+import io.quarkus.vault.transit.VerificationRequest;
+
+public interface VaultTransitSecretEngine {
+
+    // encrypt
+
+    /**
+     * Encrypt a regular string with a vault key configured in the transit secret engine.
+     * equivalent to:
+     * {@code encrypt(keyName, clearData.getBytes("UTF-8"), Optional.empty());}
+     * <p>
+     * this method is usually used in conjunction with {@link #decrypt(String, String)}
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param clearData the string to encrypt
+     * @return cipher text
+     */
+    String encrypt(String keyName, String clearData);
+
+    /**
+     * Encrypt a regular string with a vault key configured in the transit secret engine.
+     * if the key does not exist, and the policy specifies a create capability the key will be lazily created
+     * (i.e. upsert). the key can be further customized by specifying transit encryption-key configuration
+     * properties.
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param clearData the data to encrypt
+     * @param transitContext an optional context for key derivation
+     * @return cipher text
+     */
+    String encrypt(String keyName, ClearData clearData, TransitContext transitContext);
+
+    /**
+     * Encrypt a list of elements. This will return a list of results, each being either a ciphertext or an error.
+     * Each element can specify an optional context, allowing key derivation.
+     * 
+     * @see <a href="https://www.vaultproject.io/api/secret/transit/index.html#encrypt-data">encrypt data documentation</a>
+     * @param keyName name of the transit vault key
+     * @param requests the list of elements to encrypt
+     * @return a list of result objects containing either a ciphertext or an error
+     */
+    List<String> encrypt(String keyName, List<EncryptionRequest> requests);
+
+    // decrypt
+
+    ClearData decrypt(String keyName, String ciphertext);
+
+    ClearData decrypt(String keyName, String ciphertext, TransitContext transitContext);
+
+    List<ClearData> decrypt(String keyName, List<DecryptionRequest> requests);
+
+    // rewrap
+
+    String rewrap(String keyName, String ciphertext);
+
+    String rewrap(String keyName, String ciphertext, TransitContext transitContext);
+
+    List<String> rewrap(String keyName, List<RewrappingRequest> requests);
+
+    // sign
+
+    String sign(String keyName, String input);
+
+    String sign(String keyName, SignatureInput input, TransitContext transitContext);
```

Comment:
@vsevel it is my fault really, but I think it would be more correct to call it `SigningInput', because 'Signature' is the output of the sign(ing) function, sorry, please do a quick update if you agree

Reply:
@sberyozkin I agree